linera-service/tests/local_net_tests.rs

```diff
@@ -232,6 +234,86 @@ async fn test_end_to_end_reconfiguration(config: LocalNetConfig) -> Result<()> {
     Ok(())
 }
 
+/// Test if it's possible to receive epoch change messages for past epochs.
+///
+/// The epoch change messages are protected, and can't be rejected.
+#[cfg_attr(feature = "storage-service", test_case(LocalNetConfig::new_test(Database::Service, Network::Grpc) ; "storage_service_grpc"))]
+#[cfg_attr(feature = "storage-service", test_case(LocalNetConfig::new_test(Database::Service, Network::Tcp) ; "storage_service_tcp"))]
+#[cfg_attr(feature = "dynamodb", test_case(LocalNetConfig::new_test(Database::DynamoDb, Network::Grpc) ; "aws_grpc"))]
+#[cfg_attr(feature = "scylladb", test_case(LocalNetConfig::new_test(Database::ScyllaDb, Network::Tcp) ; "scylladb_tcp"))]
+#[cfg_attr(feature = "dynamodb", test_case(LocalNetConfig::new_test(Database::DynamoDb, Network::Tcp) ; "aws_tcp"))]
+#[cfg_attr(feature = "dynamodb", test_case(LocalNetConfig::new_test(Database::DynamoDb, Network::Udp) ; "aws_udp"))]
+#[test_log::test(tokio::test)]
+async fn test_end_to_end_receipt_of_old_create_committee_messages(
+    config: LocalNetConfig,
+) -> Result<()> {
+    let _guard = INTEGRATION_TEST_GUARD.lock().await;
+    tracing::info!("Starting test {}", test_name!());
+
+    let network = config.network.external;
+    let (mut net, client) = config.instantiate().await?;
+
+    let faucet_client = net.make_client().await;
+    faucet_client.wallet_init(&[], FaucetOption::None).await?;
+
+    let faucet_chain = client
+        .open_and_assign(&faucet_client, Amount::from_tokens(1_000u128))
+        .await?;
+
+    let mut faucet_service = faucet_client
+        .run_faucet(None, faucet_chain, Amount::from_tokens(2))
+        .await?;
+
+    faucet_service.ensure_is_running()?;
+
+    let faucet = faucet_service.instance();
+
+    assert_eq!(faucet.current_validators().await?.len(), 4);
+
+    client.query_validators(None).await?;
+
+    // Create a configuration for one more validator
+    net.generate_validator_config(4).await?;
+
+    // Start the validators
+    net.start_validator(4).await?;
+
+    let address = format!("{}:localhost:{}", network.short(), LocalNet::proxy_port(4));
+    assert_eq!(
+        client.query_validator(&address).await?,
+        net.genesis_config()?.hash()
+    );
+
+    // Add 4th validator
+    client
+        .set_validator(net.validator_name(4).unwrap(), LocalNet::proxy_port(4), 100)
+        .await?;
+    // client.finalize_committee().await?;
```

Comment:
I guess it works with or without that?

Reply:
Yeah, I tested it without it, but it should also happen with it. I'll confirm to be sure, but I removed it from the PR anyway.

Reply:
As discussed on Slack, this actually uncovered another bug (fixed in #2799, thanks!), so I added a separate test for it as well.